src/library/filescanner_ffmpeg.c

```diff
@@ -295,44 +310,64 @@ extract_metadata_core(struct media_file_info *mfi, AVDictionary *md, const struc
     DPRINTF(E_DBG, L_SCAN, " -> %s = %s\n", mdt->key, mdt->value);
 #endif
 
-  mdcount = 0;
-
-  /* Extract actual metadata */
-  for (i = 0; md_map[i].key != NULL; i++)
+  mdcount = 0;    
+  
+  /* Cache this search once to avoid doing it per metadata key */
+  lyricsindex = match_metadata("lyrics", md_map);
+  
+  /* Extract lyrics if any found. 
+     FFMPEG creates a metadata key that's embedding the language code in it, like 'lyrics-eng' or 'lyrics-XXX'
+     So it's not possible to query this key directly except by brute-forcing all languages.
+     Instead, we are reversing the metadata searching algorithm to query all metadata key that FFMPEG fetched 
+     and matching them against our own map. This is the most efficient method to search it without having 2 pass
+     on the KV store */
+  mdt = av_dict_get(md, "", NULL, AV_DICT_IGNORE_SUFFIX); /* from av_dict_get documentation */
+  while (mdt != NULL)
     {
-      mdt = av_dict_get(md, md_map[i].key, NULL, 0);
-      if (mdt == NULL)
-	continue;
-
-      if ((mdt->value == NULL) || (strlen(mdt->value) == 0))
-	continue;
-
-      if (md_map[i].handler_function)
-	{
-	  mdcount += md_map[i].handler_function(mfi, mdt->value);
-	  continue;
-	}
-
-      mdcount++;
-
-      if (!md_map[i].as_int)
-	{
-	  strval = (char **) ((char *) mfi + md_map[i].offset);
-
-	  if (*strval == NULL)
-	    *strval = strdup(mdt->value);
-	}
-      else
-	{
-	  intval = (uint32_t *) ((char *) mfi + md_map[i].offset);
-
-	  if (*intval == 0)
-	    {
-	      ret = safe_atou32(mdt->value, intval);
-	      if (ret < 0)
-		continue;
-	    }
-	}
+      do 
+        {
+          if ((mdt->value == NULL) || (strlen(mdt->value) == 0))
+            break;
+
+          if (memcmp(mdt->key, "lyrics-", sizeof("lyrics-") - 1) == 0)
+            i = lyricsindex; /* Replace by the generic lyrics metadata map */
+          else
+            /* Check if matches one of the mapped key */
+            i = match_metadata(mdt->key, md_map);
+            
+          if (i == -1)
+            break;
+            
+          if (md_map[i].handler_function)
+          {
+             mdcount += md_map[i].handler_function(mfi, mdt->value);
+             break;
+          }
+
+          mdcount++;
+
+          if (!md_map[i].as_int)
+            {
+              strval = (char **) ((char *) mfi + md_map[i].offset);
+
+              if (*strval == NULL)
+                *strval = strdup(mdt->value);
+            }
+          else
+            {
+              intval = (uint32_t *) ((char *) mfi + md_map[i].offset);
+
+              if (*intval == 0)
+                {
+                  ret = safe_atou32(mdt->value, intval);
+                  if (ret < 0)
+                    break;
+                }
+             }
+        }
+      while(0); /* Used so that 'break' works inside the scope */
+
+      mdt = av_dict_get(md, "", mdt, AV_DICT_IGNORE_SUFFIX);
```

Comment:
av_dict_iterate()

Reply:
Ok made the change. However, got a warning for the CI since av_dict_iterate is only on recent ffmpeg:
```bash
library/filescanner_ffmpeg.c:364:9: warning: implicit declaration of function ‘av_dict_iterate’; did you mean ‘av_codec_iterate’? [-Wimplicit-function-declaration]
  364 |   mdt = av_dict_iterate(md, NULL);
      |         ^~~~~~~~~~~~~~~
      |         av_codec_iterate
library/filescanner_ffmpeg.c:364:7: warning: assignment to ‘const AVDictionaryEntry *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  364 |   mdt = av_dict_iterate(md, NULL);
      |       ^
library/filescanner_ffmpeg.c:368:11: warning: assignment to ‘const AVDictionaryEntry *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  368 |       mdt = av_dict_iterate(md, mdt);
```

Works for me on Arch

Reply:
Ok, then we have to do a configure check for av_dict_iterate() and if it isn't present, define it from av_dict_get. I can add this.

Reply:
av_dict_iterate is just a wrapper over av_dict_get, so I'll change it back to av_dict_get, it should work everywhere without a configure check.